### PR TITLE
[Merged by Bors] - feat: `=`, `≠`, `<`, and `≤` functionality for `norm_num`

### LINKED ISSUE #1567: Inequality functionality for `norm_num`
This is an issue for tracking inequality functionality needed in `norm_num`, including `≤`, `<`, and `≠`.

For `≠`, the following example should work:
```
example {α} [DivisionRing α] [CharZero α] : (-1:α) ≠ 2 := by norm_num
```

### PR DESCRIPTION
**Pull request by thorimur**
This PR gives `norm_num` basic inequality (and equality) functionality (towards #1567), including the following:
```
example {α} [DivisionRing α] [CharZero α] : (-1:α) ≠ 2 := by norm_num
```
We implement basic logical extensions to handle `=` and `≠` together, namely `¬`, `True`, and `False`. `≠` is not given an extension, as it is handled by `=` and `¬`.

For now we only can prove `≠` for numbers given a `CharZero α` instance.

- [x] depends on: #1578

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#1578
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#1578~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: test/norm_num.lean [hunk 1]
```diff
@@ -4,216 +4,459 @@ Released under Apache 2.0 license as described in the file LICENSE.
...
+example : (2:ℝ)/4 + 4 = 3*3/2 := by norm_num1
+example : (((3:ℝ)/4)-12)<6 := by norm_num1
+example : (5:ℝ) ≠ 8 := by norm_num1
+example : (10:ℝ) > 7 := by norm_num1
+example : (2:ℝ) * 2 + 3 = 7 := by norm_num1
+example : (6:ℝ) < 10 := by norm_num1
+example : (7:ℝ)/2 > 3 := by norm_num1
+example : (4:ℝ)⁻¹ < 1 := by norm_num1
+example : ((1:ℝ) / 2)⁻¹ = 2 := by norm_num1
+-- example : 2 ^ 17 - 1 = 131071 := by norm_num1
```
Comment (gebner) [inline test/norm_num.lean:38]: Why does this no longer work?

File: test/norm_num.lean [hunk 2]
```diff
@@ -4,216 +4,459 @@ Released under Apache 2.0 license as described in the file LICENSE.
...
+-- example : (3 : ℝ) ^ (-2 : ℤ) = 1/9 := by norm_num1
 -- example : (3 : ℝ) ^ (-2 : ℤ) = 1/9 := by norm_num1
-example : (-3 : ℝ) ^ (0 : ℤ) = 1 := by norm_num
-example : (-3 : ℝ) ^ (-1 : ℤ) = -1/3 := by norm_num
--- example : (-3 : ℝ) ^ (2 : ℤ) = 9 := by norm_num
+-- example : (-3 : ℝ) ^ (0 : ℤ) = 1 := by norm_num1
+-- example : (-3 : ℝ) ^ (-1 : ℤ) = -1/3 := by norm_num1
```
Comment (gebner) [inline test/norm_num.lean:42]: Same here, why are these commented out?

File: test/norm_num.lean [hunk 3]
```diff
@@ -4,216 +4,459 @@ Released under Apache 2.0 license as described in the file LICENSE.
...
--- example : (63:ℚ) ≥ 5 := by norm_num
+example : (1:ℂ) ≠ 2 := by norm_num1
+example : (1:ℂ) / 3 ≠ 2 / 7 := by norm_num1
+
+example : (1:ℝ) ≠ 2 := by norm_num1
+
+-- example : (5 / 2:ℕ) = 2 := by norm_num1
+-- example : (5 / -2:ℤ) < -1 := by norm_num1
+-- example : (0 + 1) / 2 < 0 + 1 := by norm_num1
+-- example : Nat.succ (Nat.succ (2 ^ 3)) = 10 := by norm_num1
```
Comment (gebner) [inline test/norm_num.lean:297]: These were commented out as well, why?

File: test/norm_num.lean [hunk 4]
```diff
@@ -4,216 +4,459 @@ Released under Apache 2.0 license as described in the file LICENSE.
...
+example : 10 + 2 = 1 + 11 := by norm_num1
+-- example : 10 - 1 = 9 := by norm_num1
+-- example : 12 - 5 = 3 + 4 := by norm_num1
+-- example : 5 - 20 = 0 := by norm_num1
+-- example : 0 - 2 = 0 := by norm_num1
+-- example : 4 - (5 - 10) = 2 + (3 - 1) := by norm_num1
+-- example : 0 - 0 = 0 := by norm_num1
+-- example : 100 - 100 = 0 := by norm_num1
+-- example : 5 * (2 - 3) = 0 := by norm_num1
+-- example : 10 - 5 * 5 + (7 - 3) * 6 = 27 - 3 := by norm_num1
```
Comment (gebner) [inline test/norm_num.lean:336]: Same here.

File: test/norm_num.lean (hunk 1 above)
Comment (digama0) [reply test/norm_num.lean:38]: nat subtraction is not implemented

File: test/norm_num.lean (hunk 2 above)
Comment (digama0) [reply test/norm_num.lean:42]: zpow is not implemented

File: test/norm_num.lean (hunk 3 above)
Comment (digama0) [reply test/norm_num.lean:297]: this involves nat division, int division, and nat.succ, none of which are implemented

File: test/norm_num.lean (hunk 4 above)
Comment (gebner) [reply test/norm_num.lean:336]: I see these are nat subtraction as well.

**Comment by digama0** [comment]
bors merge

**Pull request closed without merging**